app/controllers/mixins/ems_common/metrics.rb

```diff
@@ -84,6 +84,14 @@ def refresh_or_capture_emss(task, action_name)
           params[:display] = @display
         end
       end
+
+      def queue_refresh(model)
+        ids = find_checked_items.empty? ? [checked_item_id] : find_checked_items
+        items = ids.collect { |id| find_record_with_rbac(model, id) }
+        EmsRefresh.queue_refresh(items, nil, {:create_task => true})
+        add_flash(_("Refresh successfully initiated for the selected records"))
+        render_flash
+      end
```

Comment:
I don't think that `EmsCommon::Metrics` is the right place for this.  `Core` would be better since this doesn't actually have anything to do with Metrics.

Reply:
Or you could introduce a `Mixins::EmsCommon::Refresh` module

Reply:
Yeah, `refresh_or_capture` is a terrible method you can ignore that for now we can clean it up later.